internal/utils/utils.go

```diff
@@ -13,3 +18,24 @@ func HasFinalizer(cr *metav1.ObjectMeta, name string) bool {
 	}
 	return false
 }
+
+// NormalizeLabel normalizes label string according to k8s format
+func NormalizeLabel(name string) (string, error) {
+	// convert name to lowercase and replace '.' with '-'
+	name = strings.ToLower(name)
```

Comment:
I think that upper case is allowed in label names

Reply:
You are correct. It is allowed `([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')` but it seems that all the projects that I am aware are lowercasing the keys.

Reply:
So?